frontend/public/redux.ts

```diff
@@ -53,41 +56,35 @@ const store = createStore(
   composeEnhancers(applyMiddleware(thunk)),
 );
 
-const addPluginListener = () => {
-  const reducerExtensions = pluginStore.getAllExtensions().filter(isReduxReducer);
-  const getReduxFlagsObject = () => {
-    const gatingFlags = getGatingFlagNames(reducerExtensions);
-    const featureState = store.getState()[featureReducerName];
-    return featureState ? _.pick(featureState.toObject(), gatingFlags) : null;
-  };
-
-  let flagsObject = getReduxFlagsObject();
+const pluginReducers: ReducersMapObject = {};
 
-  store.subscribe(() => {
-    const currentFlagsObject = getReduxFlagsObject();
+subscribeToExtensions<ReduxReducer>(
+  extensionDiffListener((added, removed) => {
+    removed.forEach(({ properties: { scope } }) => {
+      delete pluginReducers[scope];
+    });
 
-    if (JSON.stringify(flagsObject) !== JSON.stringify(currentFlagsObject)) {
-      flagsObject = currentFlagsObject;
+    Promise.allSettled(added.map(resolveExtension)).then((results) => {
```

Comment:
Seems like a use case for `resolveAllExtensions` utility, can be improved in a follow-up.